src/coreclr/jit/loopcloning.cpp

```diff
@@ -2044,112 +2032,93 @@ void Compiler::optCloneLoop(FlowGraphNaturalLoop* loop, LoopCloneContext* contex
     BasicBlock* h = preheader;
     // We're going to transform this loop:
     //
-    // H --> E    (if T == E, H falls through to T/E)
-    // T
-    // E
-    // B ?-> T
-    // X
+    // preheader --> header
     //
     // to this pair of loops:
     //
-    // H ?-> H3   (all loop failure conditions branch to new slow path loop head)
-    // H2 --> E   (if T == E, H2 will fall through to T/E)
-    // T
-    // E
-    // B  ?-> T
-    // X2 --> X   (only need this if B -> is conditional, not BBJ_ALWAYS)
-    // H3 --> E2  (aka slowHead. Or, H3 falls through to T2 == E2)
-    // T2
-    // E2
-    // B2 ?-> T2
-    // X
-
-    assert((h->bbFlags & BBF_LOOP_PREHEADER) != 0);
-
-    // Make a new pre-header block 'h2' for the loop. 'h' will fall through to 'h2'.
-    JITDUMP("Create new header block for loop\n");
-
-    BasicBlock* h2 = fgNewBBafter(BBJ_ALWAYS, h, /*extendRegion*/ true, /*jumpDest*/ loop->GetHeader());
-    JITDUMP("Adding " FMT_BB " after " FMT_BB "\n", h2->bbNum, h->bbNum);
-    h2->bbWeight     = h2->isRunRarely() ? BB_ZERO_WEIGHT : ambientWeight;
-    h2->bbNatLoopNum = ambientLoop;
-    h2->bbFlags |= BBF_LOOP_PREHEADER;
-
-    if (h2->JumpsToNext())
-    {
-        h2->bbFlags |= BBF_NONE_QUIRK;
-    }
-
-    assert(h->KindIs(BBJ_ALWAYS));
-    assert(h->HasJumpTo(loop->GetHeader()));
-
-    fgReplacePred(loop->GetHeader(), h, h2);
-    JITDUMP("Replace " FMT_BB " -> " FMT_BB " with " FMT_BB " -> " FMT_BB "\n", h->bbNum, loop->GetHeader()->bbNum,
-            h2->bbNum, loop->GetHeader()->bbNum);
-
-    // 'h' is no longer the loop head; 'h2' is!
-    h->bbFlags &= ~BBF_LOOP_PREHEADER;
-    optUpdateLoopHead((unsigned)(m_newToOldLoop[loop->GetIndex()] - optLoopTable), h, h2);
-
-    // Make 'h' fall through to 'h2' (if it didn't already).
-    // Don't add the h->h2 edge because we're going to insert the cloning conditions between 'h' and 'h2', and
-    // optInsertLoopChoiceConditions() will add the edge.
-    h->SetJumpDest(h2);
-    assert(h->JumpsToNext());
-    h->bbFlags |= BBF_NONE_QUIRK;
```

Comment:
I thought this part was a bit odd -- we make `h` (the original `preheader`) jump to `h2`, then insert conditions between `h` and `h2` and make `h` jump to the first condition instead (which is done inside `CondToStmtInBlock` -- seemed like something that shouldn't be its responsibility). I've updated it so that we just leave `h`/`preheader` alone until the end of this function and then directly set it to jump to the first choice condition.